lib/backend/k8s/conversion.go

```diff
@@ -337,6 +354,8 @@ func (c converter) k8sIngressRuleToCalico(r extensions.NetworkPolicyIngressRule,
 	}
 
 	// Combine desintations with sources to generate rules.
+	// TODO: This currently creates a lot of rules by making every combination of from / ports
+	// into a rule.  We can combine these so that we don't need as many rules!
```

Comment:
Yep, for the selector we'd need this I think:

```
nsSelector  || (namespace == 'x' && (podSelector || podSelector))
```

And then do a rule per-protocol using that selector.